.github/workflows/static_analysis.yml

```diff
@@ -0,0 +1,66 @@
+---
+  name: Coverity Static Analysis
+  on: [pull_request]
+  jobs:
+    polaris-scan:
+      name: Polaris Coverity Static Analysis
+      permissions:
+        packages: write
+      runs-on: ubuntu-latest
+      env:
+        VCPKG_NUGET_USER: ${{secrets.VCPKG_NUGET_USER || github.repository_owner}}
+        VCPKG_NUGET_TOKEN: ${{secrets.VCPKG_NUGET_TOKEN || secrets.GITHUB_TOKEN}}
+      steps:
+        - name: Get number of CPU cores
+          uses: SimenB/github-actions-cpu-cores@v2.0.0
+          id: cpu-cores
+
+        - name: Checkout Source
+          uses: actions/checkout@v4
+          with:
+            submodules: recursive
+            fetch-depth: 0
+
+        - name: Setup build dependencies
+          run: |
+            sudo apt-get update
+            sudo apt-get install -y gcc-10 g++-10 make mono-complete libkrb5-dev libsasl2-dev
+
+        - name: Setup VCPKG cache
+          run: |
+            . build_tooling/vcpkg_caching.sh
+            echo -e "VCPKG_BINARY_SOURCES=$VCPKG_BINARY_SOURCES
+            VCPKG_ROOT=$PLATFORM_VCPKG_ROOT" | tee -a $GITHUB_ENV
+
+        - name: Get CMake
+          uses: lukka/get-cmake@latest
+
+        - name: CMake configure
+          uses: lukka/run-cmake@v10.8
+          with:
+            cmakeListsTxtPath: ${{github.workspace}}/cpp/CMakeLists.txt
+            configurePreset: linux-debug
```

Comment:
I didn't observe many differences between debug and release in the POC, but I think we should scan release anyway